### ISSUE TITLE
`consider-using-with` (`R1732`) doesn't trigger for `Path.open()`

### ISSUE THREAD

**Issue by GideonBear** [PyCQA/pylint#7964]
### Current problem

```py
file = open('foo')          # consider-using-with
file = Path('foo').open()   # no error
path = Path('foo')
file = path.open()          # no error
```

### Desired solution

`consider-using-with` should trigger on any `Path.open` call too.
I can try to implement this if it's not too hard.

**Comment by GideonBear** [comment]
It should be added to https://github.com/PyCQA/pylint/blob/42f3cfd8f29e65b520487d44a714937925da94d9/pylint/checkers/refactoring/refactoring_checker.py#L40

**Comment by GideonBear** [comment]
@Pierre-Sassoulas Isn't this a false negative, as it's not triggering while it should be? False positive: "A message is emitted but nothing is wrong with the code". False negative: "No message is emitted but something is wrong with the code"
